### ISSUE TITLE
What about a ExpirationPolicy.ONLY_IF_ACCESSED?

### ISSUE THREAD

**Issue by mishaxz** [jhalterman/expiringmap#47]
1.  entry expires only if it has been accessed
2. this could be useful especially in conjunciton with `.maxSize` to expire entries only a set time after they have been accessed but `maxSize` prevents the map from growing out of control
3. in some unpredictable situations (like waiting for responses across the network), this could be useful when you need to keep some data in a map until you receive a confirmation request. Then you don't need it for much longer afterwards because your code is not depending upon external slowdown factors

**Comment by jhalterman** [comment]
How would this be different than `ExpirationPolicy.ACCESSED`?

**Comment by mishaxz** [comment]
I was under the impression that ACCESSED will expire entries even if they are never accessed?

**Comment by jhalterman** [comment]
That's right - so how would ONLY_IF_ACCESSED work - start the timer for an entry after it's accessed for the first time, and reset after each subsequent access?

**Comment by mishaxz** [comment]
nah, I was just thinking leave it in the map.. and if it is never accessed, it never expires. However, to prevent it from growing out of control the user could set maxSize to have it expire entries that way (just in case)

**Comment by Nohus** [comment]
This would indeed be useful, with such a policy there would be no need for any timers/threads. The expiration time would only be checked when the entry is accessed to check if it should be returned or not. It would decrease the overhead and complexity of the `ExpiringMap` while still being useful when the map size growing too big is not a concern.
